azurerm/internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -318,7 +318,7 @@ func resourceMonitorDiagnosticSettingRead(d *schema.ResourceData, meta interface
 	d.Set("eventhub_name", resp.EventHubName)
 	eventhubAuthorizationRuleId := ""
 	if resp.EventHubAuthorizationRuleID != nil && *resp.EventHubAuthorizationRuleID != "" {
-		parsedId, err := eventhubParse.NamespaceAuthorizationRuleID(*resp.EventHubAuthorizationRuleID)
+		parsedId, err := eventhubParse.NamespaceAuthorizationRuleID(strings.Replace(*resp.EventHubAuthorizationRuleID, "/AuthorizationRules/", "/authorizationRules/", 1))
```

Comment:
rather than mutating the ID here, we can use an insensitive segment parser instead, by adding a `-rewrite=true` argument to this line: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/eventhub/resourceids.go#L6 and then running `make generate` (or `go generate ./azurerm/internal/services/eventhub`, which is likely quicker here)

This then introduces a secondary `{Name}Insensitively` method which can be used for parsing the segments insensitively, meaning that we can make this:

```suggestion
		parsedId, err := eventhubParse.NamespaceAuthorizationRuleIDInsensitively(*resp.EventHubAuthorizationRuleID)
```